basic_gdeploy_doc.md

```diff
@@ -0,0 +1,298 @@
+
+###### gdeploy configuration file can be used for:
+
+    1.  [hosts] section includes ip addresses of all the machines which you want to peer probe.
+    2.  Backend setup configurations on any number of machines or nodes.
+    3.  Gluster cluster setup (Volume configuration)
+    4.  Features configurations like:
+          * NFS-Ganesha configuration
+          * Samba configuration
+            * Snapshot configuration.
+            * Geo-replication configuration.
+          * Quota configuration
+          * RH-subscription
+          * firewalld
+          * yum
+          * ctdb
+
+
+#### Installation of gdeploy:
+
+There are two ways to install gdeploy. This documentation is based on installation process
+ on `Fedora 23` on **Virtual Machine Manager**. 
+
+1.  Clone it via github repository
+2.  Using Operating System’s package manager (use rpm package manager for Fedora 23 OS)
+
+#####  Installation procedure for gdeploy using github clone:
+
+ Before we begin installing gdeploy, there are some prerequisites packages that you should 
+ already have installed on your computer to use gdeploy.
+
+The package dependencies for gdeploy may differ slightly across different operating systems 
+and environments. Other platforms and quirks will be added as and when they are encountered.
+
+This document is continuously being updated. The list of dependencies may not be exhaustive.
+ We’ll update this section periodically and frequently.
+
+
+Below is a known list of packages that you must have installed for gdeploy to work.
+ You should be able to get all these using your Operating System’s package manager.
+
+install python (No need to install pip exclusively if you are using python 2.7.10 or above)
+install pip
+install gcc
+install python-devel package
+
+These are commands that needs to be executed on `Fedora 23` VM. They should also work fine 
+if you are using any other version of Fedora.
+
+    sudo dnf -y install python-pip
+    dnf install gcc
+    dnf install redhat-rpm-config
+    dnf install python-devel
+
+Or  
+
+Run the below mentioned packages to ensure gdeploy will run.
+
+ ```yum install automake autoconf libtool flex bison openssl-devel libxml2-devel python-devel libaio-devel libibverbs-devel librdmacm-devel readline-devel lvm2-devel glib2-devel userspace-rcu-devel libcmocka-devel libacl-devel sqlite-devel redhat-rpm-config```
+
+It also depends on some other packages as well like ansible but those are 
+taken care by requirements.txt file. You'll get to see when you’ll run it.
+
+When you are done installing the above packages. We're good to go to the next step.
+
+Here are quick rundown of steps that  you need to do to install gdeploy. If you
+ are completely new to git and need some help here is the link.
+
+1.) clone the gdeploy repository by running the below command in your preferred directory.
+
+    [poo@poohostname ~]$ git clone git@github.com:gluster/gdeploy.git
```

Comment:
Modify the above paragraph as:

yum install automake autoconf libtool

Other packages are listed in requirements.txt and will be installed when pip install -r requirements.txt is run.